src/frontend/use-user.tsx

```diff
@@ -109,21 +110,22 @@ export default ({
   const [error, setError] = useState<Error | undefined>();
   const [isLoading, setIsLoading] = useState<boolean>(() => !initialUser);
 
-  useEffect((): void => {
-    if (user) return;
-
-    (async (): Promise<void> => {
-      try {
-        const response = await fetch(profileUrl);
+  const checkSession = useCallback(async (): Promise<void> => {
+    return fetch(profileUrl)
+      .then(async (response) => {
         setUser(response.ok ? await response.json() : undefined);
         setError(undefined);
-      } catch (_e) {
+      })
+      .catch(() => {
+        setUser(undefined);
         setError(new Error(`The request to ${profileUrl} failed`));
-      } finally {
-        setIsLoading(false);
-      }
-    })();
+      });
+  }, [profileUrl]);
+
+  useEffect((): void => {
+    if (user) return;
+    checkSession().finally(() => setIsLoading(false));
```

Comment:
Let's stick with using async/await everywhere so

```js
  const checkSession = useCallback(async (): Promise<void> => {
    try {
      const response = await fetch(profileUrl);
      setUser(response.ok ? await response.json() : undefined);
      setError(undefined);
    } catch (_e) {
      setUser(undefined);
      setError(new Error(`The request to ${profileUrl} failed`));
    }
  }, [profileUrl]);

  useEffect((): void => {
    if (user) return;
    (async (): Promise<void> => {
      await checkSession();
      setIsLoading(false);
    })();
  }, [user]);
```

It's easier to debug and passing promises into promises like `promise.then(async () => {})` is generally considered an anti pattern